### ISSUE TITLE
In DefaultCodegen.java, CodegenParameter.is<TYPE>Param flags should be set before invoking postProcessParameter

### ISSUE THREAD

**Issue by neilotoole** [swagger-api/swagger-codegen#2561]
In [DefaultCodegen#fromOperation](https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultCodegen.java#L1554), we call `fromParameter()` to build the `CodegenParameter` object, and then a few lines [later](https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultCodegen.java#L1581) we check the type of the param, and set the type booleans, e.g. `p.isQueryParam = new Boolean(true)`, etc. These type flags should probably be set inside `fromParameter()`, as that method calls out to `postProcessParameter()`, which is where a language-specific generator can modify the parameter.

TLDR: the `p.is<Type>Param` flags should be set before `postProcessParameter()` is invoked.

**Comment by wing328** [comment]
PR merged into master
